verifyFiles.py

```diff
@@ -153,19 +159,14 @@ def n_differences_across_subjects(conditions_dict,root_dir,metrics):
                     else:
 			abs_path_c=os.path.join(root_dir,c,subject,file_name)
                         abs_path_d=os.path.join(root_dir,d,subject,file_name)
-                        # File sizes are identical: compute the checksums
-	                if checksums_flag:
-                            folder_c_checksums_file=os.path.join(root_dir,c,subject,"checksums-after.txt")
-			    folder_d_checksums_file=os.path.join(root_dir,d,subject,"checksums-after.txt")
-			    #print folder_c_checksums_file,folder_d_checksums_file
-                            if ((os.path.isfile(folder_c_checksums_file) and os.path.isfile(folder_d_checksums_file))):
-				if not ((os.path.isdir(abs_path_c) and os.path.isdir(abs_path_d))):
-    			            if not is_checksum_equal(folder_c_checksums_file,folder_d_checksums_file,subject,file_name):
-				        diff[key][file_name]+=1
-                                        files_are_different=True                            
+			file_name_checksum = "exec/"+subject+"/"+file_name
+    			if checksums_flag:
+			#  if not ((os.path.isdir(abs_path_c) and os.path.isdir(abs_path_d))):
+			    if (checksums_from_file_dict[c][subject][file_name_checksum] != checksums_from_file_dict[d][subject][file_name_checksum]):
+		                diff[key][file_name]+=1
+                                files_are_different=True                            
                         elif checksum(abs_path_c) != checksum(abs_path_d): # TODO:when they are multiple conditions, we will compute checksums multiple times.We should avoid that.
```

Comment:
Your code misses the case where both checksums are identical on the local machine but are different from what is in the checksum file. The condition tested here should be:
```
if checksum(abs_path_c) != checksums_from_file_dict[c][subject][file_name_checksum] 
   log_error("Checksum of abs_path_c in checksum file is different from what is computed here." 
if checksum(abs_path_d) != checksums_from_file_dict[d][subject][file_name_checksum])
     log_error("Checksum of abs_path_d in checksum file is different from what is computed here." 
```